src/APIM_ARMTemplate/apimtemplate/Extractor/EntityExtractors/TagExtractor.cs

```diff
@@ -69,6 +95,45 @@ public async Task<Template> GenerateTagsTemplateAsync(string apimname, string re
                     templateResources.Add(TagTemplateResource);
                 }
             }
+        }
+
+        /// <summary>
+        /// Creates resource Template
+        /// </summary>
+        /// <param name="apimname"></param>
+        /// <param name="resourceGroup"></param>
+        /// <param name="singleApiName"></param>
+        /// <param name="apiTemplateResources"></param>
+        /// <param name="productTemplateResources"></param>
+        /// <param name="policyXMLBaseUrl"></param>
+        /// <param name="policyXMLSasToken"></param>
+        /// <returns>The contexts for an ARM template</returns>
+        /// <remarks>
+        /// This method starts with the request URL the way that it was originally built, determines the tags associations, and builds the ARM resources.
+        /// Additionally, if "nextLink" in the API results is not null, loops back to repeat the process with the results from the "nextLink" query.
+        /// Exits the while loop once no more results are available.
+        /// </remarks>
```

Comment:
you can pass extractor instance to make parameters fewer

Reply:
That is actually the public interface that ExtractorUtils::GenerateTemplates calls (line 102 in ExtractorUtils.cs).  If I change that, then I'm breaking the interface.  Alternatively, I could also change ExtractorUtils but then this particular Extractor would have a different argument set than the other extractors.

Reply:
Yes, in order to make this happen, we need to refactor some parts of the code

Reply:
I've updated the changes.  I tested it on my original error case.  I should have thought to do it that way.  It seems counter-intuitive to "skip 0", but the code is much cleaner.  Thank you @RupengLiu.

Reply:
You are welcome!